native/src/main/java/io/ballerina/persist/cmd/Generate.java

```diff
@@ -0,0 +1,252 @@
+            errStream.println(e.getMessage());
+        }
+    }
+
+    private ArrayList<Entity> readBalFiles() throws BalException {
+        ArrayList<Entity> returnMetaData = new ArrayList<>();
+        Path dirPath = Paths.get(this.sourcePath);
+        List<Path> fileList;
+
+
+        try (Stream<Path> walk = Files.walk(dirPath)) {
+            if (walk != null) {
+                fileList = walk.filter(Files::isRegularFile).collect(Collectors.toList());
+                if (hasSyntacticDiagnostics(Paths.get(this.sourcePath)) ||
+                        hasSemanticDiagnostics(Paths.get(this.sourcePath), this.projectEnvironmentBuilder)) {
+                    throw new BalException("Error occurred while building the project!");
+                }
+                for (Path filePath : fileList) {
+                    if (filePath.toString().endsWith(".bal")) {
+                        String[] pathElements = filePath.toString().strip().split(Pattern.quote(File.separator));
+                        String module = "";
+                        String[] dirElements = this.sourcePath.split(Pattern.quote(File.separator));
+                        if (pathElements.length < 2) {
+                            module = "";
+                        } else if (!Arrays.asList(dirElements).contains(pathElements[pathElements.length - 2])) {
+                            module = pathElements[pathElements.length - 2];
+                        }
+                        ArrayList<Entity> retData = BalSyntaxTreeGenerator.readBalFiles(filePath, module);
+                        if (retData.size() != 0) {
+                            returnMetaData.addAll(retData);
+                        }
+                    }
+                }
+                return returnMetaData;
+            }
+        } catch (IOException e) {
+            throw new BalException("Error occurred while reading bal files!");
+        }
+        return new ArrayList<>();
+    }
+
+    private void generateClientBalFile(Entity entity) throws BalException {
+        SyntaxTree balTree = BalSyntaxTreeGenerator.generateBalFile(entity);
+        String clientPath;
+        if (entity.getModule().equals("")) {
+            clientPath = Paths.get(this.sourcePath, "modules", "generated_clients",
+                    entity.getEntityName().toLowerCase() + "_client.bal").toAbsolutePath().toString();
+        } else {
+            clientPath = Paths.get(this.sourcePath, "modules", "generated_clients",
+                            entity.getModule() + "_" + entity.getEntityName().toLowerCase() + "_client.bal")
+                    .toAbsolutePath().toString();
+        }
+        try {
+            writeOutputFile(balTree, clientPath);
+        } catch (IOException e) {
+            throw new BalException("Error occurred in formatting ballerina client object!");
+        } catch (FormatterException e) {
+            throw new BalException("Error occurred while generating ballerina client object!");
```

Comment:
are these error messages correct? please check.

Reply:
it should be a file not an object.

Reply:
`IOException` is thrown when there is a error in writing the file and `FormatterException` is thrown when `content = Formatter.format(syntaxTree.toSourceCode());` failed. So the error content is correct, but the style and wording can be improved